File: app/src/main/java/com/github/quentin7b/sn/boot/StartUpReceiver.java
package com.github.quentin7b.sn.boot;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Start a service that show up notifications when the phone is ready to go
 */
public class StartUpReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            // Start the service to show notifications
            context.startService(new Intent(context, StartUpService.class));
        }
    }
}

File: app/src/main/kotlin/com/github/quentin7b/sn/boot/StartUpReceiver.kt
package com.github.quentin7b.sn.boot

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Start a service that show up notifications when the phone is ready to go
 */
class StartUpReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_BOOT_COMPLETED == intent.action) {
            // Start the service to show notifications
            context.startService(StartUpService.newIntent(context))
        }
    }
}
